common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,152 @@
+        self.user = UserFactory(is_staff=True)
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+        self.course = CourseFactory()
+        self.entitlements_list_url = reverse('entitlements_api:v1:entitlements-list')
+        self.entitlements_detail_path = 'entitlements_api:v1:entitlements-detail'
+
+    def _get_data_set(self, user, course_uuid):
+        """
+        Get a basic data set for an entitlement
+        """
+        return {
+            "user": user.username,
+            "mode": "verified",
+            "course_uuid": course_uuid,
+            "order_number": "EDX-1001"
+        }
+
+    def test_auth_required(self):
+        self.client.logout()
+        response = self.client.get(self.entitlements_list_url)
+        assert response.status_code == 401
+
+    def test_staff_user_required(self):
+        not_staff_user = UserFactory()
+        self.client.login(username=not_staff_user.username, password=UserFactory._DEFAULT_PASSWORD)
+        response = self.client.get(self.entitlements_list_url)
+        assert response.status_code == 403
+
+    def test_add_entitlement_with_missing_data(self):
+        entitlement_data_missing_parts = self._get_data_set(self.user, str(uuid.uuid4()))
+        entitlement_data_missing_parts.pop('mode')
+        entitlement_data_missing_parts.pop('course_uuid')
+
+        response = self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data_missing_parts),
+            content_type='application/json',
+        )
+        assert response.status_code == 400
+
+    def test_add_entitlement(self):
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+
+        response = self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        assert response.status_code == 201
+        results = response.data
+
+        course_entitlement = CourseEntitlement.objects.get(
+            user=self.user,
+            course_uuid=course_uuid
+        )
+
+        assert course_entitlement is not None
+        assert results['uuid'] == str(course_entitlement.uuid)
+        assert str(course_entitlement.course_uuid) == str(course_uuid)
+        assert course_entitlement.enrollment_course_run is None
+        assert course_entitlement.mode == 'verified'
+        assert course_entitlement.created is not None
+        assert course_entitlement.user.username == self.user.username
+
+    def test_get_entitlements(self):
+        entitlements = CourseEntitlementFactory.create_batch(2)
+
+        response = self.client.get(
+            self.entitlements_list_url,
+            content_type='application/json',
+        )
+        assert response.status_code == 200
+
+        results = response.data.get('results', [])
+        assert results[0]['uuid'] == str(entitlements[0].uuid)
```

Comment:
Take a look at how we validate output in the Discovery API. These three lines can be simplified: `assert response.data == CourseEntitlementSerializer(entitlements, multiple=True).data`

In order to fully test the code, we first verify the serializers behave as expected: https://github.com/edx/course-discovery/blob/master/course_discovery/apps/api/tests/test_serializers.py. Once we have proven our serializers are valid, we can use them within the view tests to validate the views' responses. Thus, outside of special cases, you can simply compare `response.data` to the serialization of the model instances you expect to be returned.

If your tests are only checking specific fields, you're going to miss something if/when the fields are updated.